extensions/ql-vscode/src/test-adapter.ts

```diff
@@ -185,12 +188,39 @@ export class QLTestAdapter extends DisposableObject implements TestAdapter {
 
     this._testStates.fire({ type: 'started', tests: tests } as TestRunStartedEvent);
 
+    const currentDatabaseUri = this.databaseManager.currentDatabaseItem?.databaseUri;
+    const databasesUnderTest = this.databaseManager.databaseItems
+      .filter(database => tests.find(testPath => database.isAffectedByTestsUnder(testPath)));
+    await withProgress(
```

Comment:
Darn. I was a little mistaken here and I think your original approach was better.

`this.runningTask` is a `CancellationTokenSource` and `withProgress` gives you a `CancellationToken`. The latter allows you to cancel long running operations and listen for when cancellation happens. The former is for creating and disposing these tokens.

I was hoping we could use a single token and progress monitor for the entire test running operation. This would let users cancel some tests partway through a run.

What you have now creates a first progress monitor, which could be cancelled but does not end the entire operation. And when the tests are running, there is no progress monitor to cancel at all (but the red _stop_ button still works).

There might be a way to get all these operations to work in one operation, but that is a bigger refactor. I think just reverting to what you had before with empty progress monitors is fine.

Reply:
Thanks!  That makes sense.  Even with the withProgress() call, I cannot get the progress monitor window to show up at all, so it is not much of a loss anyway.